### ISSUE TITLE
Add color status and gutters for VCS (Git)

### ISSUE THREAD

**Issue by wyrmling** [godotengine/godot-proposals#1089]
**Describe the project you are working on:**
Several game projects with decent amount of scenes/scripts. Some of them are being developed in team. All of them stored in Git-based repositories.

**Describe the problem or limitation you are having in your project:**
My use-case - an incremental approach, and relies on finishing small steps and commiting them to Git repo, thus "cleaning" the overall project view. During development I'm frequently using Git status gutters in PhpStorm to track new / refactored / removed code.
In opposite to this - Godot IDE doesn't allow me to get this info. Sometimes I prefer to switch to VS Code for coding, which has Godot plugin. But still for me it's a temporary solution, as I have to constantly switch back to IDE.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
As it was mentioned in issue godotengine/godot#25442 it should be an integrated solution, to really solve the above mentioned user experience.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
As for me - best examples would be VSC & JetBrains IDE functionality:
#### VS Code
![defold_vcs_gutter_vscode](https://user-images.githubusercontent.com/13387159/41196122-31409042-6c42-11e8-8cf9-ab2b98f9d286.png)

#### IntelliJ IDEA
![defold_vcs_gutter_idea](https://user-images.githubusercontent.com/13387159/41196120-3117eb24-6c42-11e8-9875-5a3a068e7599.png)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I don't think there are a "few line solution" for this. Also - suggested enhancement will be used on constant basis.

**Is there a reason why this should be core and not an add-on in the asset library?:**
Yes, I'm aware of https://github.com/godotengine/godot-git-plugin/
The main reason why it has to be added to core - it's meant to be mainly informational feature.
Despite functionality that it may suggest - I personally don't need it, as prefer to interact with Git via CLI or, in complicated cases, something UI-based, like SmartGit.

**Comment by Calinou** [comment]
> The main reason why it has to be added to core - it's meant to be mainly informational feature (plus "revert block" function).

The feature was specifically intended to be provided by plugins as it's designed to support multiple version control systems, not just Git. Otherwise, none of the added complexity is needed :slightly_smiling_face:

**Comment by pycbouh** [comment]
@Calinou But shouldn't the base for it be provided in core, so that all VCS plugins can use it? I'm not sure how concerns are separated between VCS plugins and the core, but this looks like a part that must be provided by core in an abstracted and easy to use by plugins way.

**Comment by wyrmling** [comment]
> The feature was specifically intended to be provided by plugins as it's designed to support multiple version control systems, not just Git.

I agree, all VCS are important, that's was only a way to show that Git has bigger priority from my POV. I will remove "Git" from the title.

Though, I still think that it's significantly important to have integrated VCS status gutters in IDE on constants basis, as we can't imagine more or less serious project not using some VCS - it's really helping to not get lost in your code.
Also, it will help a lot in team development.

**Comment by Calinou** [comment]
There's a plugin adding Git modified line gutter highlights: https://github.com/Datoh/godot-diff-margin

The implementation is rather small and self-contained: https://github.com/Datoh/godot-diff-margin/blob/main/addons/diff-margin/plugin.gd

**Comment by blurymind** [comment]
> elf

doesnt seem to work on 4.6
